.github/workflows/eden_gcp.yml

```diff
@@ -0,0 +1,112 @@
+---
+name: EdenGCP
+on:  # yamllint disable-line rule:truthy
+  workflow_run:
+    workflows: ["publish"]
+    branches: [main]
+    types:
+      - completed
+# yamllint disable rule:line-length
+jobs:
+  integration:
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        hv: ["kvm", "xen"]
+    steps:
+      - name: get eden
+        uses: actions/checkout@v2
+        with:
+          repository: 'lf-edge/eden'
+      - name: fetch or build eve
+        env:
+          TAG: pr${{ github.event.pull_request.number }}
+          CACHE: evebuild/danger
+        run: |
+          if docker pull "$CACHE:$TAG-${{ matrix.hv }}"; then
+             docker tag  "$CACHE:$TAG-${{ matrix.hv }}" "lfedge/eve:$TAG-${{ matrix.hv }}"
+             docker tag  "$CACHE:$TAG-${{ matrix.hv }}" "lfedge/eve:$TAG-${{ matrix.hv }}-amd64"
+          else
+            echo "No tag $CACHE:$TAG-${{ matrix.hv }} found"
+          fi
+          echo "TAG=$TAG" >> $GITHUB_ENV
+      - name: Public IP
+        id: ip
+        uses: actions/http-client
+        result-encoding: string
+        script: |
+          const maxRetries = parseInt(core.getInput('maxRetries'), 10);
+          const http = new HttpClient('haythem/public-ip', undefined, { allowRetries: true, maxRetries: maxRetries });
+          try {
+            const ipv4 = await http.getJson<IPResponse>('https://api.ipify.org?format=json');
+          } catch (error) {
+            echo ("Didn't get anything")
+            echo(error.message)
+          }
+          return ipv4
+      - name: setup packages
+        run: |
+          sudo apt update
```

Comment:
small nit -- if you're going to change this workflow anyway -- I've noticed that you can drop apt update part -- just to save time